filesystem/src/main/java/com/nytimes/android/external/fs/filesystem/BreadthFirstFileTreeIterator.java

```diff
@@ -51,7 +52,9 @@
      * @param root The root directory
      */
     BreadthFirstFileTreeIterator(@Nonnull File root) {
```

Comment:
Since we got this wrong the first time let's wrap the logic in tests to make sure it won't blow up again.